rfc/20241023-rfc-tracking-issues.md

```diff
@@ -0,0 +1,209 @@
+
+Under this proposal, a decision to revise an RFC would be represented either as a blocker or as an item of work in the RFC's tracking issue, and then a new PR linked to the tracking issue proposes the amendments directly to the existing document, replacing previous content as appropriate. If the proposed amendements are approved then the new text becomes the authority on what work is planned and the project owner should update the tracking issue as needed to reflect the new set of work.
+
+Once an RFC's tracking issue has been closed to represent that work on it is complete, the RFC content is frozen as a historical record of that work. If we later plan other work in the same area -- which might in some cases conflict with what was previously proposed and accepted -- we should write a new RFC that describes its proposed changes relative to the outcome of the previous RFC.
+
+One exception to the frozen state: if a new RFC is proposed that materially changes a decision made and recorded in an earlier accepted RFC, the PR for the new RFC should add a note in each relevant part of the older RFC that mentions that the decision has been changed and links to the new RFC, so that it's clear to a future reader of the old RFC that part of it has been obsoleted retroactively. Consider using [GitHub Markdown's "alert" extensions](https://github.blog/changelog/2023-12-14-new-markdown-extension-alerts-provide-distinctive-styling-for-significant-content/) to clearly separate the later-added content from the original frozen content, and add the new markers in the same PR that introduces the new RFC so that the changes are more strongly connected in the history.
+
+The "chains" of linked RFCs that might be caused by the frozen state could potentially become quite long over time for product areas that experience frequent change, but thankfully older RFCs tend to become less and less relevant over time. Any significant feature of the product or its codebases should also be documented in some separate location in a "how it currently is" shape, rather than a "how we changed it over time" shape, so that it isn't necessary to piece together information on currently-available features by trawling through the RFC history unless the historical change decisions _themselves_ are the point of interest.
+
+Some existing locations for longer-lived supporting documentation include:
+
+- [The OpenTofu docs](https://opentofu.org/docs/), maintained in the `website` directory in the root of the main OpenTofu repository, for end-user oriented documentation.
+- The `docs` directory in the root of the main OpenTofu repository, for developer-oriented descriptions of system architecture and the design of non-trivial features.
```

Comment:
I think the question of how to organize docs is probably better saved for another RFC; my goal here was only to give some examples of locations where we already keep documentation to try to clarify what kinds of documentation this section was talking about.

As the person who originally put this `docs` directory here I can say that my intent with it was specifically _that_ it is tethered to particular version branches and particular commits, so that if you're working on something for an old release you can more easily find the docs about the codebase _as it existed in that release_. Of course, that's only true if we actually remember to update the docs when we change something, and so I suppose this section doubles as a reminder to actually do that. :grinning: 

However, I understand that there are different tradeoffs here and would be happy to discuss other ways to organize the developer-oriented documentation in a separate RFC focused on that topic!